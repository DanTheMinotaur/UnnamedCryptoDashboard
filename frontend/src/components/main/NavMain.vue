<script setup lang="ts">
import routes from '../../router/routes'
</script>

<template>
  <nav class="sidenav navbar navbar-vertical navbar-expand-xs border-0 border-radius-xl my-3 fixed-start ms-3   bg-gradient-dark" id="sidenav-main">
    <hr class="horizontal light mt-0 mb-2">
    <div class="collapse navbar-collapse  w-auto " id="sidenav-collapse-main">
      <ul class="navbar-nav">
        <li v-for="route in routes" :key="route.path" :to="route.path" class="nav-item">
            <RouterLink class="nav-link text-white" :to="route.path" :active-class="'bg-gradient-primary active'">
                <div class="text-white text-center me-2 d-flex align-items-center justify-content-center">
                    <i class="material-icons opacity-10">{{ route.meta.icon }}</i>
                </div>
                <span class="nav-link-text ms-1">{{ route.name }}</span>
            </RouterLink>
        </li>
      </ul>
    </div>
  </nav>
</template>