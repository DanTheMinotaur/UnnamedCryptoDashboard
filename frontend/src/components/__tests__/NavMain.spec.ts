import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import NavMain from '../main/NavMain.vue' // Assuming your component file is named Sidebar.vue

describe('Sidebar', () => {
  it('renders correctly with routes', async () => {
    // Mock routes data
    const routes = [
      { path: '/dashboard', name: 'Dashboard', meta: { icon: 'home' } },
      { path: '/holding', name: 'Holdings', meta: { icon: 'info' } },
      // Add more mock routes if needed
    ]

    // Mount the component with mocked routes
    const wrapper = mount(NavMain, {
      props: { routes }
    })

    // Wait for Vue to finish rendering
    await wrapper.vm.$nextTick()

    // Assert that the component renders correctly
    expect(wrapper.html()).toMatchSnapshot()
  })

  // Add more tests as needed
})