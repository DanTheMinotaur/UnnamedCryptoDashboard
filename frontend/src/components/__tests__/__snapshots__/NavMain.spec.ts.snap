// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > renders correctly with routes 1`] = `
"<nav class="sidenav navbar navbar-vertical navbar-expand-xs border-0 border-radius-xl my-3 fixed-start ms-3 bg-gradient-dark" id="sidenav-main" routes="[object Object],[object Object]">
  <hr class="horizontal light mt-0 mb-2">
  <div class="collapse navbar-collapse w-auto" id="sidenav-collapse-main">
    <ul class="navbar-nav">
      <li to="/" class="nav-item">
        <routerlink class="nav-link text-white" to="/" active-class="bg-gradient-primary active">
          <div class="text-white text-center me-2 d-flex align-items-center justify-content-center"><i class="material-icons opacity-10">dashboard</i></div><span class="nav-link-text ms-1">Dashboard</span>
        </routerlink>
      </li>
      <li to="/holdings" class="nav-item">
        <routerlink class="nav-link text-white" to="/holdings" active-class="bg-gradient-primary active">
          <div class="text-white text-center me-2 d-flex align-items-center justify-content-center"><i class="material-icons opacity-10">wallet</i></div><span class="nav-link-text ms-1">Holdings</span>
        </routerlink>
      </li>
    </ul>
  </div>
</nav>"
`;
